src/mono/mono/metadata/assembly.c

```diff
@@ -1530,12 +1533,16 @@ mono_assembly_open_from_bundle (MonoAssemblyLoadContext *alc, const char *filena
 	MonoImage *image = NULL;
 	MONO_ENTER_GC_UNSAFE;
 	gboolean is_satellite = culture && culture [0] != 0;
-
-	if (is_satellite)
-		image = open_from_satellite_bundle (alc, filename, status, culture);
-	else
-		image = open_from_bundle_internal (alc, filename, status, FALSE);
-
+	const char *name;
+	if (is_satellite) {
+		name = g_strdup (filename);
```

Comment:
No need to duplicate the name in this case.

Reply:
bumping up the free into the scope of the else block